proposed/2022/globalpackagereference.md

```diff
@@ -0,0 +1,125 @@
+# Global Package Reference for Central Package Management
+
+- [Jeff Kluge](https://github.com/jeffkl)
+- 2022-05-05
+- [[Feature]: CPVM equivalent of GlobalPackageReference? (#10159)](https://github.com/NuGet/Home/issues/10159)
+- GitHub PR (GitHub PR link)
+
+## Summary
+<!-- One-paragraph description of the proposal. -->
+A common pattern in repositories is to have every project in the tree reference packages that are used for development purposes only.  This includes
+packages that provide assembly versioning, signing, code analysis, etc.  We want to provide a built-in mechanism for users to define packages that
+are used by every project but only for development purposes.
+
+## Motivation 
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+Users can manually reference the package in a common import like Directory.Build.props but there is no built-in safe guards to ensure that they don't
+accidentally consume more assets than just development assets.  For example, you don't want to have a package's compile-time assets be referenced by
+every assembly in your repository.  Instead, individual projects should express their build time dependencies and project references should receive
+them transitively.  But you should not need to add a package reference in order to have your whole tree use code analysis or assembly versioning.
+
+## Explanation
+
+### Functional explanation
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+We'll introduce a new MSBuild item group named `<GlobalPackageReference />` which will define packages that every project should reference but has
+default metadata to ensure that only package assets related to development purposes are used.
+
+**NOTE:** This functionality will only be enabled if a user has opt-ed into [Central Package Management (CPM)](https://docs.microsoft.com/nuget/consume-packages/central-package-management).
+
+```xml
+<Project>
+  <ItemGroup>
+    <GlobalPackageReference Include="Nerdbank.GitVersioning" Version="1.0.0" />
+  </ItemGroup>
+</Project>
+```
+
+### Technical explanation
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+`<GlobalPackageReference />` will just be syntactical sugar for a `<PackageReference />` with the following metadata:
+
+* IncludeAssets = `Analyzers;Build;BuildMultitargeting;BuildTransitive`
```

Comment:
Note that this is *different* from how `DevelopmentDependency` works in PackageReference.
If you read that thread, you'll discover that the feature is a `personal favorite` of mine :D 

If I could break things, I'd say development dependencies:

* Can't bring compile/runtime assets
* Have to be self contained, and can't have dependencies
* Are not part of the standard project graph, something theoretically addressed by not allowing them to have dependencies.

Note that what I described is effectively the guidance for writing msbuild task packages.

Reply:
See details: https://github.com/NuGet/Home/pull/11794

Reply:
I'm going to remove all references to "development dependency" so there is no confusing.  This feature is about making sure repo-wide packages only consume _build_ assets.